src/udev/udev-node.c

```diff
@@ -481,45 +481,116 @@ static int link_update(sd_device *dev, const char *slink_in, bool add) {
         return -ELOOP;
 }
 
-int udev_node_update_old_links(sd_device *dev, sd_device *dev_old) {
-        const char *name;
+static int xsprintf_dev_num_path_from_sd_device(sd_device *dev, char **ret) {
```

Comment:
maybe call this `device_get_devpath_by_devnum()`?

i.e. this is in many ways like device_get_depth(), but returns the major/minor path, instead of the primary device node path.